crypto/encode_decode/encoder_lib.c

```diff
@@ -240,6 +256,57 @@ int OSSL_ENCODER_CTX_add_extra(OSSL_ENCODER_CTX *ctx,
     return 1;
 }
 
+int OSSL_ENCODER_CTX_prune_encoders(OSSL_ENCODER_CTX *ctx)
```

Comment:
Why is this API call needed as a separate call? Isn't it leaking unnecessarily implementation details? Why were the non-matching encoders added in the first place?

Reply:
I've been asking myself the same thing...  I'm currently dealing with the decoders and didn't have to do that there, so I'm going to revisit this soon.